xbmc/addons/gui/GUIDialogAddonSettings.cpp

```diff
@@ -128,7 +132,134 @@ bool CGUIDialogAddonSettings::ShowForAddon(const ADDON::AddonPtr& addon,
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10013));
+          item->SetProperty("id", BASE_SETTINGS);
+          itemsGeneral.Add(item);
+        }
+        {
+          // Label: "Add add-on configuration"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10014));
+          item->SetProperty("id", ADD_INSTANCE);
+          itemsGeneral.Add(item);
+        }
+        {
+          // Label: "Delete add-on configuration"
+          const CFileItemPtr item = std::make_shared<CFileItem>(g_localizeStrings.Get(10015));
+          item->SetProperty("id", DELETE_INSTANCE);
+          itemsGeneral.Add(item);
+        }
+
+        std::vector<uint32_t> ids = addon->GetKnownInstanceIds();
+        uint32_t highestId = 0;
+        for (const auto& id : ids)
+        {
+          // Format label: "Add-on configuration #{0:d}"
+          const std::string label = StringUtils::Format(g_localizeStrings.Get(10017), id);
+          const CFileItemPtr item = std::make_shared<CFileItem>(label);
+          item->SetProperty("id", id);
+          itemsInstances.Add(item);
+          if (id > highestId)
+            highestId = id;
+        }
+
+        for (auto& it : itemsGeneral)
+          dialog->Add(*it);
+
+        itemsInstances.Sort(SortByLabel, SortOrderAscending);
+        for (auto& it : itemsInstances)
+          dialog->Add(*it);
+
+        dialog->SetButtonFocus(true);
+        dialog->Open();
+
+        if (dialog->IsButtonPressed())
+          return false;
+
+        if (dialog->IsConfirmed())
+        {
+          const CFileItemPtr& item = dialog->GetSelectedFileItem();
+          instanceNumber = item->GetProperty("id").asInteger();
+          if (instanceNumber < GENERAL_BUTTON_START)
+          {
+            break;
+          }
+          else if (instanceNumber == ADD_INSTANCE)
+          {
+            instanceNumber = highestId + 1;
+            addon->GetSettings(instanceNumber);
+            addon->SaveSettings(instanceNumber);
+            byNew = true;
+            break;
+          }
+          else if (instanceNumber == DELETE_INSTANCE)
+          {
+            dialog->Reset();
+            dialog->SetHeading(10023); // Label: "Select add-on configuration to remove"
+            dialog->SetUseDetails(false);
+            for (auto& it : itemsInstances)
+              dialog->Add(*it);
+            dialog->SetButtonFocus(true);
+            dialog->Open();
+            if (dialog->IsButtonPressed())
+              continue;
+
+            if (dialog->IsConfirmed())
+            {
+              const CFileItemPtr& item = dialog->GetSelectedFileItem();
+              instanceNumber = item->GetProperty("id").asInteger();
+
+              // Format label: "Are you sure to remove the selected add-on configuration #{0:d} settings?"
+              const auto text = StringUtils::Format(g_localizeStrings.Get(10010), instanceNumber);
```

Comment:
We should display the configuration's title here, not the id, which says nothing to the user